hibernate-core/src/main/java/org/hibernate/boot/model/internal/EmbeddableBinder.java

```diff
@@ -388,7 +388,15 @@ static Component fillEmbeddable(
 
 			final XProperty property = propertyAnnotatedElement.getProperty();
 			if ( property.isAnnotationPresent( GeneratedValue.class ) ) {
-				processGeneratedId( context, component, property );
+				if ( isIdClass || subholder.isOrWithinEmbeddedId() ) {
+					processGeneratedId( context, component, property );
+				}
+				else {
+					LOG.warn(
+							"Property " + property.getName() + " of "
+									+ getPath( propertyHolder, inferredData )
+									+ " is annotated with @GeneratedValue but is not part of an identifier, the @GeneratedValue annotation will be ignored" );
+				}
```

Comment:
This `WARN` should be a `MappingException` or `AnnotationException`.

Reply:
I agree, while previously we were ignoring the annotation it's best to make it clear the mapping is not valid to avoid confusion (as in the case of this Jira).

Reply:
changed to throw an `AnnotationException`

Reply:
Perfect, thanks, but the message still contains "`, the @GeneratedValue annotation will be ignored`" which should now be removed :)

I propose just:

```java
" is annotated '@GeneratedValue' but is not part of an identifier"
```

Reply:
Oh, actually, I would also like to see quotes around the interpolates bits, i.e.

```
"Property '" + property.getName() + "' of '"
	+ getPath( propertyHolder, inferredData )
	+ "' is annotated '@GeneratedValue' but is not part of an identifier"
```

Reply:
done